test/test_instrument/NDB_BVL_Instrument_testtest.class.inc

```diff
@@ -42,6 +42,9 @@ class NDB_BVL_Instrument_testtest extends NDB_BVL_Instrument
 
         // setup the form
         $this->_setupForm();
+
+        // set data entry type
+        $this->DataEntryType = 'DirectEntry';
```

Comment:
Why is this set to DirectEntry? As far as I know this test instrument isn't used as a survey instrument

Reply:
I believe it is needed because the following logic was added in this PR and it affects testing.
@kongtiaowang can maybe provide more details

```
            if ($this->preview !== true && $this->DataEntryType !== 'DirectEntry') {
                $this->freeze();
            }
```

Reply:
Hi @driusan ,the integration test of testtest instrument will test the direct input information. I added DirectEntry since @Jesscall have modifed the DataEntryType rule.

Reply:
You have the following 5 conditions that would make the instrument be frozen
1. Instrument is NOT in progress
2. the user does not have data_entry permission
3. the user does not have access to the timepoint at which the instrument is instantiated
4. the instrument is not in preview mode
5. the instrument is not a direct entry

Why choose to solve the problem with option 5 ? I would say options 1, 2 and 3 are your easy logical solutions !

Reply:
I think Rida means "make the instrument not be frozen". Assuming he does, I agree. "Direct entry" means survey mode and doesn't get frozen because when it's set, the participant is entering data directly (after the entry is complete they can no longer access the URL..)

If the test doing data entry on an instrument that was supposed to be frozen and is now broken because the freeze bug is fixed, the solution should be to fix the test and/or data, not fake being a survey instrument for testing non-survey functionality in order to work around the freeze call.

Reply:
Thanks for the feedback. 
I agree with Dave, it's likely an issue with the test doing data entry on an instrument meant to be frozen and this fix is messing with that. In that case the test case should be updated. 

Pushing Rida's recent feedback and will monitor the test suite logs for indications of where the issue might be.